src/redteam/providers/advNoise.ts

```
@@ -0,0 +1,322 @@
+ */
+function addTypos(text: string, typoRate: number = 0.1): string {
+  const chars = text.split('');
+  const numTypos = Math.max(1, Math.floor(text.length * typoRate));
+
+  for (let i = 0; i < numTypos; i++) {
+    const pos = Math.floor(Math.random() * chars.length);
+    const typoType = Math.floor(Math.random() * 3);
+
+    switch (typoType) {
+      case 0: // Insert random character
+        const randomChar = String.fromCharCode(97 + Math.floor(Math.random() * 26)); // a-z
```

Biome (1.9.4)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Unsafe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)